samples/net/azure_iot_hub/boards/nrf7002dk_nrf5340_cpuapp_ns.conf

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# General
+CONFIG_POSIX_CLOCK=y
+CONFIG_HW_ID_LIBRARY=y
+CONFIG_HW_ID_LIBRARY_SOURCE_NET_MAC=y
+CONFIG_DK_LIBRARY=y
+
+# Heap and stacks
+CONFIG_MAIN_STACK_SIZE=4096
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
+CONFIG_HEAP_MEM_POOL_SIZE=153600
+CONFIG_NET_RX_STACK_SIZE=2048
+
+# Set newlib C to prevent build error complaining that string.h is not supported
+CONFIG_NEWLIB_LIBC_FLOAT_PRINTF=y
+
+# Wi-Fi
+CONFIG_WIFI=y
+CONFIG_WIFI_NRF700X=y
+CONFIG_WIFI_NRF700X_LOG_LEVEL_ERR=y
+CONFIG_WIFI_MGMT_EXT=y
+CONFIG_WIFI_CREDENTIALS=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+
+# Shell
+CONFIG_SHELL=y
+CONFIG_SHELL_STACK_SIZE=4096
+
+# WPA
+CONFIG_WPA_SUPP=y
+CONFIG_WPA_SUPP_LOG_LEVEL_ERR=y
+
+# Zephyr NET Connection Manager Connectivity layer.
+CONFIG_L2_WIFI_CONNECTIVITY=y
+CONFIG_L2_WIFI_CONNECTIVITY_AUTO_CONNECT=n
+CONFIG_L2_WIFI_CONNECTIVITY_AUTO_DOWN=n
+
+# DNS
+CONFIG_DNS_RESOLVER=y
+
+# NET sockets
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_L2_ETHERNET=y
+CONFIG_NET_TCP=y
+CONFIG_NET_TCP_WORKQ_STACK_SIZE=2048
+CONFIG_NET_UDP=y
+CONFIG_NET_SOCKETS_OFFLOAD=n
+CONFIG_NET_DHCPV4=y
+CONFIG_NET_CONTEXT_SNDTIMEO=y
+CONFIG_NET_CONTEXT_RCVTIMEO=y
+
+# TLS networking
+CONFIG_NET_SOCKETS_SOCKOPT_TLS=y
+CONFIG_NET_SOCKETS_ENABLE_DTLS=n
+CONFIG_NET_SOCKETS_TLS_MAX_CONTEXTS=2
+
+# nRF Security
+CONFIG_NRF_SECURITY=y
+CONFIG_PSA_CRYPTO_DRIVER_OBERON=n
+CONFIG_CC3XX_BACKEND=y
+
+# mbedTLS
+CONFIG_MBEDTLS=y
+CONFIG_MBEDTLS_ENABLE_HEAP=y
+CONFIG_MBEDTLS_HEAP_SIZE=120000
+CONFIG_MBEDTLS_RSA_C=y
+
+# FOTA
+CONFIG_AZURE_FOTA=y
+CONFIG_AZURE_FOTA_APP_VERSION_AUTO=n
+CONFIG_AZURE_FOTA_TLS=y
+# Change the security tag to the tag where the certificates are provisioned
+# for the server where the FOTA image is hosted
+CONFIG_AZURE_FOTA_SEC_TAG=11
+CONFIG_BOOTLOADER_MCUBOOT=y
+CONFIG_FOTA_DOWNLOAD=y
+CONFIG_DFU_TARGET=y
+CONFIG_MCUBOOT_IMG_MANAGER=y
+CONFIG_IMG_MANAGER=y
+CONFIG_STREAM_FLASH=y
+CONFIG_IMG_ERASE_PROGRESSIVELY=y
+CONFIG_DOWNLOAD_CLIENT=y
+CONFIG_DOWNLOAD_CLIENT_STACK_SIZE=4096
+
+# Enable external flash to host MCUBoot secondary partition
+CONFIG_SPI=y
+CONFIG_SPI_NOR=y
+CONFIG_SPI_NOR_SFDP_DEVICETREE=y
+CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK=y
+
+# TFM
+CONFIG_TFM_PROFILE_TYPE_SMALL=y
```

Comment:
The nRF Connect SDK only supports CONFIG_TFM_PROFILE_TYPE_NOT_SET and CONFIG_TFM_PROFILE_TYPE_MINIMAL.
See https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/security/tfm.html#minimal-build

Reply:
As discussed offline, using `CONFIG_TFM_PROFILE_TYPE_NOT_SET` and `CONFIG_TFM_PROFILE_TYPE_MINIMAL` will result in several KConfig options in the sample. Due to this we prefer to use `CONFIG_TFM_PROFILE_TYPE_SMALL` and be happy with the configurations it provides for now and fine tune this later (if need be).